app/src/main/java/pl/pwr/zpi/bcycle/mobile/TripCreationActivity.kt

```diff
@@ -0,0 +1,161 @@
+        val DESCRIPTION_KEY = "DESCRIPTION_KEY"
+        val formatterDateTime = DateTimeFormatter.ofPattern(DATE_TIME_FORMAT).withZone(ZoneId.systemDefault())
+    }
+    
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_trip_creation)
+        setListeners()
+        supportActionBar?.hide()
+    }
+
+    private fun setListeners() {
+        bt_next.setOnClickListener {
+            if(isFormCorrect()){
+                startActivity(createIntentAndSaveData())
+            }
+        }
+        bt_start_date.setOnClickListener{chooseDateTime(tv_start_time, tv_start_date, true)
+        }
+        bt_finish_date.setOnClickListener{chooseDateTime(tv_finish_time, tv_finish_date, false)
+        }
+    }
+
+    private fun createIntentAndSaveData() : Intent{
+        val intent = Intent(this, TripCreationMapActivity::class.java)
+        val bundle = Bundle()
+        intent.putExtra(NAME_KEY,  et_name.text.toString())
+
+
+        /*intent.putExtra(START_DATE_KEY, tv_start_date.text.toString())
+        intent.putExtra(START_TIME_KEY, tv_start_time.text.toString())
+        intent.putExtra(END_DATE_KEY, tv_finish_date.text.toString())
+        intent.putExtra(END_TIME_KEY, tv_finish_time.text.toString())*/
+
+        bundle.putSerializable(START_DATE_KEY,ZonedDateTime.parse(tv_start_date.text.toString() + " " +  tv_start_time.text.toString(), formatterDateTime))
+        bundle.putSerializable(END_DATE_KEY,ZonedDateTime.parse( tv_finish_date.text.toString() + " " +  tv_finish_time.text.toString(), formatterDateTime))
+
+        intent.putExtra(DESCRIPTION_KEY,  et_desc.text.toString())
+        intent.putExtras(bundle)
+        return intent
+    }
+
+    private fun chooseDateTime(timeTV:TextView, dateTV:TextView, startDate:Boolean) {
+        val cal = Calendar.getInstance()
+        val timeSetListener = TimePickerDialog.OnTimeSetListener { _, hour, minute ->
+            cal.set(Calendar.HOUR_OF_DAY, hour)
+            cal.set(Calendar.MINUTE, minute)
+            timeTV.text = SimpleDateFormat(TIME_FORMAT).format(cal.time)
+        }
+        val dateSetListener = DatePickerDialog.OnDateSetListener { _, year, month, day ->
+            cal.set(Calendar.YEAR, year)
+            cal.set(Calendar.MONTH, month)
+            cal.set(Calendar.DAY_OF_MONTH, day)
+            dateTV.text = SimpleDateFormat(DATE_FORMAT).format(cal.time)
+        }
+        TimePickerDialog(
+            this,
+            timeSetListener,
+            cal.get(Calendar.HOUR_OF_DAY),
+            cal.get(Calendar.MINUTE),
+            true
+        ).show()
+        DatePickerDialog(
+            this,
+            dateSetListener,
+            cal.get(Calendar.YEAR),
+            cal.get(Calendar.MONTH),
+            cal.get(Calendar.DAY_OF_MONTH)
+        ).show()
+
+        if(startDate){
+            bt_start_date.setMargins(rightMarginDp = 80)
```

Comment:
The UI mockup has a label and text boxes, it would be better IMO to keep it like that instead of changing the UI like this (perhaps as label + text view + button to edit).